### PR TITLE
TST: Adds test coverage for pd.read_json index type

### LINKED ISSUE #28558: pd.read_json With convert_axes Produces Different Index Type than Empty Frame
Another edge case brought up by #28510 . I'm not sure if we have a definite stance on this and there are probably a few other issues that are similar, though couldn't find anything exactly matching this

Note the index types for an empty data frame

```python
>>> df = pd.DataFrame()
>>> df.index
Index([], dtype='object')
>>> df.columns
Index([], dtype='object')
```

In `read_json` with `convert_axes` the axes labels end up with a different dtype

```python
>>> for orient in ["columns", "index", "records", "values", "split"]:
...     print(pd.read_json("{}", orient=orient, convert_axes=True).columns)
...
Float64Index([], dtype='float64')
Float64Index([], dtype='float64')
Float64Index([], dtype='float64')
Float64Index([], dtype='float64')
Float64Index([], dtype='float64')
```

### PR DESCRIPTION
**Pull request by mcgeestocks**
- [x] closes #28558 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

## Description: 
@mroeschke mentioned that the merged fix for issue [#28558](https://github.com/pandas-dev/pandas/issues/28558) could use a [test](https://github.com/pandas-dev/pandas/issues/28558#issuecomment-1528187186). This PR adds the missing test for index type equality between the `DataFrame` and `read_json` functions.

### CODE REVIEW

File: pandas/tests/io/json/test_pandas.py [hunk 1]
```diff
@@ -1398,6 +1398,15 @@ def test_read_json_table_dtype_raises(self, dtype):
...
             read_json(dfjson, orient="table", dtype=dtype)
 
+    @pytest.mark.parametrize("orient", ["index", "columns", "records", "values"])
+    def test_read_json_table_empty_dtype(self, orient):
+        # GH 28558
+
+        df = DataFrame()
+        dfjson = read_json("{}", orient=orient, convert_axes=True)
+
+        assert df.index.dtype == dfjson.index.dtype
```
Comment (mroeschke) [inline pandas/tests/io/json/test_pandas.py:1408]: Could you compare the entire index?

```python
dfjson = read_json("{}", orient=orient, convert_axes=True)
result = df.json.index
expected = df.index
tm.assert_index_equal(result, expected)
```

Same for the `.columns`

Comment (mcgeestocks) [reply pandas/tests/io/json/test_pandas.py:1408]: Will update with your suggestions shortly!

File: pandas/tests/io/json/test_pandas.py [hunk 2]
```diff
@@ -1398,6 +1398,16 @@ def test_read_json_table_dtype_raises(self, dtype):
...
 
+    @pytest.mark.parametrize("orient", ["index", "columns", "records", "values"])
+    def test_read_json_table_empty_axes_dtype(self, orient):
+        # GH28558
+
+        expected = DataFrame()
+        result = read_json("{}", orient=orient, convert_axes=True)
+
+        tm.assert_index_equal(result.index, expected.index)
+        tm.assert_equal(result.columns, expected.columns)
```
Comment (mroeschke) [inline pandas/tests/io/json/test_pandas.py:1409]:
```suggestion
        tm.assert_index_equal(result.columns, expected.columns)
```

**Comment by mcgeestocks** [comment]
Updated with your latest suggestion @mroeschke. Let me know if there is anything else.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @mcgeestocks

**Pull request merged**